src/PhpBrew/VariantBuilder.php

```diff
@@ -561,6 +562,39 @@ public function __construct()
             }
         };
 
+        /*
+         * --with-sodium option
+         *
+         * --with-sodium=shared
+         * --with-sodium=[dir]
+         *
+         * On ubuntu you need to install libsodium-dev
+         */
+        $this->variants['sodium'] = function (Build $build, $val = null) {
+            if ($val) {
+                return "--with-sodium=$val";
+            }
+
+            if ($prefix = Utils::findIncludePrefix('sodium.h')) {
+                return "--with-sodium=$prefix";
+            }
+
+            if ($prefix = Utils::getPkgConfigPrefix('sodium')) {
+                return "--with-sodium=$prefix";
+            }
+
+            $possiblePrefixes = array('/usr/local/opt/sodium');
+            $foundPrefixes = array_filter($possiblePrefixes, 'file_exists');
+            if (count($foundPrefixes) > 0) {
+                return '--with-sodium='.$foundPrefixes[0];
+            }
+
+            // This will create sodium.so file for dynamic loading.
+            echo 'Compiling with sodium=shared, please install libsodium-dev or sodium header files if you need';
```

Comment:
This is misleading. The implementation doesn't configure it as shared. Also, the message about dependencies doesn't cover all supported OS'es and is prone to become invalid over time, so it's better to move it into [Wiki](https://github.com/phpbrew/phpbrew/wiki/Variant-Dependencies).